core/src/main/java/org/kiwix/kiwixmobile/core/utils/SimpleRecyclerViewScrollListener.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2021 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core.utils
+
+import androidx.recyclerview.widget.RecyclerView
+
+class SimpleRecyclerViewScrollListener(
+  private val onLayoutScrollListener: (RecyclerView, Int) -> Unit // here we are calling callback
+) :
+  RecyclerView.OnScrollListener() {
+  override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
+    super.onScrollStateChanged(recyclerView, newState)
+    onLayoutScrollListener(
+      recyclerView,
+      newState
+    ) // implement the callback by passing recyclerview and  newState
+    super.onScrollStateChanged(recyclerView, newState)
+  }
+}
+
+
+
```

Comment:
You should remove these blank lines that's why static analyses are failing.

Reply:
done.